### ISSUE TITLE
Fetching formats failed

### ISSUE THREAD

**Issue by Rafee-M** [jgreco/mpv-youtube-quality#28]
When pressing Ctrl + f gives out an error

![image](https://user-images.githubusercontent.com/69535896/151718411-3b1e5739-5335-4194-bea8-523a22bc35f0.png)

**Comment by benyaminl** [comment]
I also experience the same. does this only works on linux?

I think this problem raise because we users use ytdlp, and at line 196-210 on .lua script is really depends on the youtube-dl, lead to broken downloading format. hmm.. Let me try to see if changing to yt-dlp.exe works

EDIT: Yes it works if we replace youtube-dl to yt-dlp on line 164. 

Is there anyway to bind this config to .conf file?

@jgreco ? as Youtube-dl already abandoned by the author itself.. and spin off as yt-dlp?
Prove :
lua script
![image](https://user-images.githubusercontent.com/2817332/151915692-aa31daa5-86fc-4237-9561-6c9ce9a82ea1.png)
the GUI result 
![image](https://user-images.githubusercontent.com/2817332/151915723-6691ce35-cbf4-4a3a-9d68-900b0e3e7128.png)

**Comment by Rafee-M** [comment]
@benyaminl I've changed it and I still get the same error

**Comment by benyaminl** [comment]
Does yt-dlp is included on windows env PATH? If no the error will raise..
so please try to check it. Check also if yoy are using ytdl or ytdlp.




Regards,
Benyamin Limanto
sent from my Sony XPERIA XZ Premium

**Comment by RafeeDaBoy** [comment]
Yes, it is in windows environment PATH as it is in the same directory as mpv and yt-dlp works in the command line

![image](https://user-images.githubusercontent.com/69535896/152030683-0419aff9-ad19-4bd5-af41-2745267f396f.png)

**Comment by Rafee-M** [comment]
Moving it from the mpv's install directory to ~\AppData\Roaming\mpv\scripts solved the issue, along with changing youtube-dl to yt-dlp on line 164. Thanks!

**Comment by benyaminl** [comment]
@RafeeDaBoy I think yt-dlp path should be tested on cmd first, because I've yt-dlp and mpv on same folder and I put it on env path, then it works. Please remember that when you change env path, you need to restart, as lua/python can't detect automatically

I think this issue should kept open as we need to adopt it as conf. 

ping @jgreco

**Comment by Rafee-M** [comment]
Yes, I had checked if yt-dlp and mpv was working in cmd and it was.

And I've reopened the issue.

**Comment by arktnld** [comment]
Made this code to fix this error

```lua
-- Function to check if a command is executable
function isExecutable(command)
    local f = io.popen("which " .. command)
    local path = f:read("*a")
    f:close()
    return string.len(path) > 0
end

-- Check if youtube-dl or yt-dlp is in PATH
if isExecutable("youtube-dl") then
    yt_tool = "youtube-dl"
elseif isExecutable("yt-dlp") then
    yt_tool = "yt-dlp"
end

local ytdl = {
    path = yt_tool,
    searched = false,
    blacklisted = {}
}

```

**Comment by benyaminl** [comment]
@arktnld I think you need to make a PR, so @jgreco could see it. Hopefully it will be updated. Thanks for the heads up @arktnld

**Comment by arktnld** [comment]
@benyaminl done  #32

**Comment by subhanshu1111** [comment]
> Made this code to fix this error
> 
> ```lua
> -- Function to check if a command is executable
> function isExecutable(command)
>     local f = io.popen("which " .. command)
>     local path = f:read("*a")
>     f:close()
>     return string.len(path) > 0
> end
> 
> -- Check if youtube-dl or yt-dlp is in PATH
> if isExecutable("youtube-dl") then
>     yt_tool = "youtube-dl"
> elseif isExecutable("yt-dlp") then
>     yt_tool = "yt-dlp"
> end
> 
> local ytdl = {
>     path = yt_tool,
>     searched = false,
>     blacklisted = {}
> }
> ```
where do i add this inside my lua script? at the end of file? 
edit: I added your lua script from pull request you made but still doesnt work on macos

**Comment by arktnld** [comment]
>  edit: I added your lua script from pull request you made but still doesnt work on macos

@subhanshu1111 First, test if these commands are working:
```sh
youtube-dl -F {URL}
```
or
```sh
yt-dlp -F {URL}
```
Sometimes youtube-dl or yt-dlp don't work correctly.

**Comment by IlgazC** [comment]
> 

I am having this problem on openSUSE TW and I also removed the unmaintained youtube-dl and use yt-dl only which is in /usr/bin 

Here is the result of -F
 yt-dlp -F https://www.youtube.com/watch?v=P-8C-f1flcw
[youtube] Extracting URL: https://www.youtube.com/watch?v=P-8C-f1flcw
[youtube] P-8C-f1flcw: Downloading webpage
[youtube] P-8C-f1flcw: Downloading ios player API JSON
[youtube] P-8C-f1flcw: Downloading android player API JSON
[youtube] P-8C-f1flcw: Downloading m3u8 information
[youtube] P-8C-f1flcw: Downloading m3u8 information
[youtube] P-8C-f1flcw: Downloading m3u8 information
[info] Available formats for P-8C-f1flcw:
ID  EXT RESOLUTION FPS │   TBR PROTO │ VCODEC        VBR ACODEC     MORE INFO
─────────────────────────────────────────────────────────────────────────────
233 mp4 audio only     │       m3u8  │ audio only        unknown    Default
234 mp4 audio only     │       m3u8  │ audio only        unknown    Default
269 mp4 256x144     15 │  290k m3u8  │ avc1.42C00B  290k video only
91  mp4 256x144     15 │  290k m3u8  │ avc1.42c00b       mp4a.40.5
229 mp4 426x240     30 │  546k m3u8  │ avc1.4D4015  546k video only
92  mp4 426x240     30 │  546k m3u8  │ avc1.4d4015       mp4a.40.5
230 mp4 640x360     30 │ 1210k m3u8  │ avc1.4D401E 1210k video only
93  mp4 640x360     30 │ 1210k m3u8  │ avc1.4d401e       mp4a.40.2
231 mp4 854x480     30 │ 1569k m3u8  │ avc1.4D401F 1569k video only
94  mp4 854x480     30 │ 1569k m3u8  │ avc1.4d401f       mp4a.40.2
232 mp4 1280x720    30 │ 2969k m3u8  │ avc1.4D401F 2969k video only
95  mp4 1280x720    30 │ 2969k m3u8  │ avc1.4d401f       mp4a.40.2
